mage/query-modules/python/conditional-execution.md

```diff
@@ -0,0 +1,177 @@
+
+[![docs-source](https://img.shields.io/badge/source-conditional_execution-FB6E00?logo=github&style=for-the-badge)](https://github.com/memgraph/mage/blob/main/python/do.py)
+
+| Trait              | Value                                                 |
+| ------------------ | ----------------------------------------------------- |
+| **Module type**    | <Highlight color="#FB6E00">**module**</Highlight>     |
+| **Implementation** | <Highlight color="#FB6E00">**Python**</Highlight>     |
+| **Parallelism**    | <Highlight color="#FB6E00">**sequential**</Highlight> |
+
+## Procedures
+
+:::info
+
+Using the following procedures to run queries that execute global operations is
+currently not supported and returns a warning.
+The operations in question are:
+
+* index creation/deletion
+* constraint creation/deletion
+* changing the isolation level globally
+* setting the storage mode
+
+:::
+
+### `case(conditionals, else_query, params)`
+
+Given a list of condition-query pairs, `do.case` executes the query associated
+with the first condition evaluating to `true` (or the `else query` if none are
+`true`) with the given parameters.
+
+Parameters are prefixed with `$` like `$param_name`. For examples, see
+[here](https://memgraph.com/docs/cypher-manual/other-features#parameters).
+
+#### Input:
+
+* `conditionals: List[Any]` ➡ Variable-length list of condition-query pairs
+  structured as `[condition, query, condition, query, …​]`. Conditions are
+  `boolean` and queries are `string`.
+* `else_query: string (default = "")` ➡ The query to be executed if no
+  condition evaluates to true.
+* `params: Map (default = NULL)` ➡ If any of the given queries is parameterized,
+  provide a `{param_name: param_value}` map to be applied to them.
+
+#### Output:
+
+* `value: Map` ➡ Contains the result record of the executed query. Each `value` corresponds to one result record.
+
+#### Usage:
+
+```cypher
+MATCH (n)
+WITH size(collect(n)) as n_nodes
+CALL do.case([n_nodes = 0,
+              "RETURN 'empty' AS graph_status;",
+              n_nodes = 1,
+              "RETURN 'one_node' AS graph_status;"],
+              "RETURN 'multiple nodes' AS graph_status;")
+YIELD value
+RETURN value.graph_status AS graph_status;
+```
+
+### `when(condition, if_query, else_query, params)`
+
+`do.when` evaluates the given condition and executes the `if query` or the
+`else query` depending on whether the condition is satisfied.
+
+Parameters are prefixed with `$` like `$param_name`. For examples, see
+[here](https://memgraph.com/docs/cypher-manual/other-features#parameters).
+
+#### Input:
+
+* `condition: boolean` ➡ Determines what query to execute.
+* `if_query: string` ➡ The query to be executed if the condition is satisfied.
+* `else_query: string (default = "")` ➡ The query to be executed if the
+  condition isn’t satisfied.
+* `params: Map (default = NULL)` ➡ If `if_query` or `else_query` parameterized,
+   provide a `{param_name: param_value}` map to be applied.
```

Comment:
```suggestion
   provide a `{param_name: param_value}` map to be applied.
```

that will be applied?

Reply:
Both are valid constructions and unambiguous, but feel free to make this change if you think it’s more understandable :)